src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1151,27 +1151,91 @@ MonodroidRuntime::monodroid_dlopen (const char *name, int flags, char **err, [[m
 {
 	unsigned int dl_flags = monodroidRuntime.convert_dl_flags (flags);
 	bool libmonodroid_fallback = false;
-
+	bool name_is_full_path = false;
+	bool name_needs_free = false;
 	/* name is nullptr when we're P/Invoking __Internal, so remap to libxa-internal-api */
-	if (name == nullptr) {
+	if (name == nullptr || strstr (name, "xa-internal-api") != nullptr) {
+#if defined (WINDOWS)
+		char *tmp_name = nullptr;
+
+		auto probe_dll_at = [&](const char *the_path) -> bool {
+			const char *last_sep = strrchr (the_path, MONODROID_PATH_SEPARATOR_CHAR);
+			if (last_sep != nullptr) {
+				tmp_name = utils.strdup_new (the_path, last_sep - the_path);
+				tmp_name = utils.string_concat (tmp_name, MONODROID_PATH_SEPARATOR, API_DSO_NAME);
```

Comment:
This is a memory leak.  The preceding line `utils.strdup_new()` does a `new char[…]`, and `utils.string_concat()` *also* returns a `new char[…]` *without* a `delete[] s1`.  Consequently, this sequence of statements "loses" the first `new char[…]` held by `tmp_name`, causing it to leak memory.